src/Rings/mpoly-localizations.jl

```diff
@@ -92,9 +92,37 @@ Base.iterate(U::MPolyPowersOfElement, i::Int) = (i < length(U.a) ? (U.a[i+1], i+
 is_trivial(U::MPolyPowersOfElement) = (U == units_of(ambient_ring(U)))
 
 ### printing
+function Base.show(io::IO, ::MIME"text/plain", S::MPolyPowersOfElement)
+  io = pretty(io)
+  println(io, "Multiplicative subset")
+  print(io, Indent())
+  println(io, "of ", Lowercase(), ambient_ring(S))
+  print(io, "given by the powers of ")
+  print(io, "[")
+  for (i,s) in enumerate(denominators(S))
+    print(io, s)
+    if i!=length(denominators(S))
+      print(io, ", ")
+    end
+  end
```

Comment:
Not too important, but all those could be simplified by writing `print(io, "["); join(io, denominators(S), ", "); print(io, "]")`